usage.go

```diff
@@ -208,6 +207,22 @@ func (a *Application) UsageForContextWithTemplate(context *ParseContext, indent
 			return string(c)
 		},
 	}
+
+	// kingpin originally uses "github.com/alecthomas/template" which is a fork
+	// of go 1.4's "text/template" with one additional feature to remove
+	// newlines after "}}\".
+	//
+	// "text/template" is added back in this file to utilize the latest
+	// features of go. To be backwards compatible with "}}\" in existing
+	// templates, "}}\" is replaced with " -}}" to remove the new lines.
+	//
+	// One side effect of " -}}" is it may swallow extra whitespaces on the
+	// next line, where "}}\" will keep them. To make sure the behavior is the
+	// same as before, an action block {{" "}} is used to replace the first
+	// space of the next line to stop the swallowing.
+	tmpl = strings.ReplaceAll(tmpl, "}}\\\n ", " -}}\n{{\" \"}}")
```

Comment:
Could we remove the extra logic and migrate templates to `-}}`, or does that break too much stuff?

Reply:
without `}}\`, it will adds many random new lines here and there. I was even attempted to fork "text/template" and apply the `}}\`. Took a quick look and finds it will take a lot of effort. So maybe another day.

@codingllama i am happy to reverse to the original change while addressing a few of your comments. Let me do that and discard this. Thanks a lot for reviewing!